dotcom-rendering/src/web/layouts/FrontLayout.tsx

```diff
@@ -76,6 +55,100 @@ const isToggleable = (
 	} else return index != 0 && !isNavList(collection);
 };
 
+const getMerchHighPosition = (
+	collectionCount: number,
+	isNetworkFront: boolean | undefined,
+) => {
+	if (collectionCount < 4) {
+		return 2;
+	} else if (isNetworkFront) {
+		return 5;
+	} else {
+		return 4;
+	}
+};
+
+/**
+ * On mobile, we remove the first container if it is a thrasher
+ * and remove a container if it, or the next sibling, is a commercial container
+ * we also exclude any containers that are directly before a thrasher
+ * then we take every other container, up to a maximum of 10, for targeting MPU insertion
+ */
+
+const getMobileAdPositions = (
+	isNetworkFront: boolean | undefined,
+	collections: DCRCollectionType[],
+) => {
+	const merchHighPosition = getMerchHighPosition(
+		collections.length,
+		isNetworkFront,
+	);
+
+	const positions: (number | undefined)[] = collections
+		.map((collection, collectionIndex) => {
+			const isThrasher = collection.collectionType === 'fixed/thrasher';
+			const isFirst = collectionIndex === 0;
+			const isNearMerchandising =
+				collectionIndex === merchHighPosition ||
+				collectionIndex + 1 === merchHighPosition;
+			const isNearThrasher =
+				collections[collectionIndex + 1]?.collectionType ===
+				'fixed/thrasher';
+			if (isFirst && isThrasher) return false;
+			if (isNearMerchandising) return false;
+			if (isNearThrasher) return false;
+			else if (
+				collectionIndex % 2 === 0 &&
+				collectionIndex < collections.length - 1
+			) {
+				return true;
+			}
+			return false;
+		})
+		.map((shouldDisplayAd, collectionIndex) =>
+			shouldDisplayAd ? collectionIndex : undefined,
+		)
+		.filter((index) => typeof index === 'number')
```

Comment:
If you turn this into a type predicate, you won't have to assign the type type of positions and it will be `number[]`

```suggestion
		.filter((index): index is number => typeof index === 'number')
```